blocks/identity-block/components/social-sign-on/utils/useSocialSignIn.js

```diff
@@ -135,20 +134,21 @@ function useSocialSignIn(
 
 	useEffect(() => {
 		const checkIsLoggedIn = () => {
-			setTimeout(() => {
+			const timeout = setTimeout(() => {
 				Identity.isLoggedIn().then((isLoggedIn) => {
 					if (isLoggedIn) {
-						const validatedURL = validateURL(redirectURL);
-						window.location = validatedURL;
+						window.location = validateURL(redirectURL);
 					}
 				});
 			}, 1500);
+  			return () => clearTimeout(timeout)
```

Comment:
This only returns the function to the calling code (159 below) and will not return that to the useEffect...

Since 159/161 are in functions themselves, they will also not return this clean up code to the useEffect.  In this case, you need a shared `timeout` pointer defined at the top of the useEffect, which you clean up at the end of the useEffect.
```
useEffect(() => {
  let timeout;
  const checkIsLoggedIn ...
    timeout = setTimeout(...
  };
  ...
  // rest of code
  ...
  return () => clearTimeout(timeout);
}, [ ... ]);
```
Make sense?

Reply:
this should be okay since only one of those timeouts will occur.